Changing the protocol check in the resize method

keslerm opened the pull request:
Tested w/ ruby 2.2.1 - was causing a crash with INVALID URI when attempting to resize a local image with a space in the path name.

This is a simpler check and should prevent that.

sdsykes commented:
I think it's better to rescue URI::InvalidURIError instead of testing the start of the string. (So we don't fail on "httpmyimage.jpg").

keslerm commented:
I updated the code to include the :// portion of the string which should prevent any false matches like you mentioned.

I thought about trying to rescue around the URI and defaulting to local but then that would make that try/catch part of the control flow instead of simply error handling.

sdsykes commented:
Ok, use input =~ URI.regexp(['http','https','ftp']) then.

keslerm commented:
Okay I changed it to URI.regexp like you suggested, did not know about that piece of URI and doesn't look documented on the ruby docs, also did not know about the =~ operator. So that was very helpful (still very new to ruby).

This passes all tests including the one i updated (the ico test was reporting FormatNotSupported instead of UnknownImageType)

sdsykes commented:
Looks good, thanks!

sdsykes commented:
Oh, one more thing before I merge - did you want to write a test for the issue this patch fixes?

keslerm commented:
I just added the test for that scenario.
